components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -416,6 +427,64 @@ protected JWTClaimsSet createJWTClaimSet(OAuthAuthzReqMessageContext authAuthzRe
         return jwtClaimsSet;
     }
 
+    /**
+     * To get the scope of the token to be added to the JWT claims.
+     *
+     * @param authAuthzReqMessageContext Auth Request Message Context.
+     * @param tokenReqMessageContext     Token Request Message Context.
+     * @param jwtClaimsSet               Relevant JWT Claim Set.
+     * @return scope of token.
+     */
+    private String getScope(OAuthAuthzReqMessageContext authAuthzReqMessageContext,
+            OAuthTokenReqMessageContext tokenReqMessageContext, JWTClaimsSet jwtClaimsSet) {
+
+        String[] scope;
+        String scopeString = null;
+        if (tokenReqMessageContext != null) {
+            scope = tokenReqMessageContext.getScope();
+        } else {
+            scope = authAuthzReqMessageContext.getApprovedScope();
+        }
+        if (ArrayUtils.isNotEmpty(scope)) {
+            scopeString = OAuth2Util.buildScopeString(scope);
+            if (log.isDebugEnabled()) {
+                log.debug("Scope exist for the jwt access token with subject " + jwtClaimsSet.getSubject() + " and "
```

Comment:
Why not get the subject from the context?
We are building the jwt claimset multiple times unnecessarily just to get context info for logs. IMO we can use information from the context instead

Reply:
Changed with the commit 63aa142358c853c194a1d0035fb99fcaf16ac479